src/librustc_mir/transform/borrow_check.rs

```diff
@@ -0,0 +1,1287 @@
+
+                                    // (base initialized; no need to
+                                    // recur further)
+                                    break;
+                                }
+                                _ => {}
+                            }
+                        }
+                    }
+
+                    lvalue = base;
+                    continue;
+                }
+            }
+        }
+    }
+
+    fn check_for_conflicting_loans(&mut self,
+                                   context: Context,
+                                   _location: Location,
+                                   _bk: BorrowKind,
+                                   lvalue_span: (&Lvalue<'gcx>, Span),
+                                   flow_state: &InProgress<'b, 'gcx>) {
+        // NOTE FIXME: The analogous code in old borrowck
+        // check_loans.rs is careful to iterate over every *issued*
+        // loan, as opposed to just the in scope ones.
+        //
+        // (Or if you prefer, all the *other* iterations over loans
+        // only consider loans that are in scope of some given
+        // CodeExtent)
+        //
+        // The (currently skeletal) code here does not encode such a
+        // distinction, which means it is almost certainly over
+        // looking something.
+        //
+        // (It is probably going to reject code that should be
+        // accepted, I suspect, by treated issued-but-out-of-scope
+        // loans as issued-and-in-scope, and thus causing them to
+        // interfere with other loans.)
+        //
+        // However, I just want to get something running, especially
+        // since I am trying to move into new territory with NLL, so
+        // lets get this going first, and then address the issued vs
+        // in-scope distinction later.
+
+        let state = &flow_state.borrows;
+        let data = &state.base_results.operator().borrows();
+
+        debug!("check_for_conflicting_loans location: {:?}", _location);
+
+        // does any loan generated here conflict with a previously issued loan?
+        for (g, gen) in state.elems_generated().map(|g| (g, &data[g])) {
+            for (i, issued) in state.elems_incoming().map(|i| (i, &data[i])) {
+                debug!("check_for_conflicting_loans gen: {:?} issued: {:?} conflicts: {}",
+                       (g, gen, self.base_path(&gen.lvalue),
+                        self.restrictions(&gen.lvalue).collect::<Vec<_>>()),
+                       (i, issued, self.base_path(&issued.lvalue),
+                        self.restrictions(&issued.lvalue).collect::<Vec<_>>()),
+                       self.conflicts_with(gen, issued));
+                if self.conflicts_with(gen, issued) {
+                    self.report_conflicting_borrow(context, lvalue_span, gen, issued);
+                }
+            }
+        }
+
+        // does any loan generated here conflict with another loan also generated here?
```

Comment:
Hmm. This loop was meant to be analogous to the one [here](https://github.com/rust-lang/rust/blob/1b6c960/src/librustc_borrowck/borrowck/check_loans.rs#L362) in AST borrowck.

I was starting from an assumption that in general, a given statement in a control-flow graph could introduce multiple loans (and those can be mutually conflicting).

But now that you have questioned the presence of this code, I am experiencing some deja-vu, in that I am realizing once again that MIR is designed such that each borrow arises solely from an `Rvalue::Ref`, each of which in turn arises solely from a `StatementKind::Assign`  (perhaps that is what you were getting at by pointing out that a borrow "is a mutate_lvalue" ...?), and thus there is *at most one* loan generated from any given point in the control flow graph.

So, that's good news; not only can this loop go away, but also the supporting code like the  `IdxSet::pairs` method...